### DISCUSSION TITLE
Autocomplete preline

### DISCUSSION THREAD

**Discussion by TuvikCarsales** [htmlstreamofficial/preline#364] [General]
I can't find an example with https://preline.co/docs/combobox.html#basic-usage. An example there with api connection. The field is apiUrl. What if I want to call the api before calling the element? What if I want to put it as a variable? I must admit that the documentation is very limited and bad. It doesn't include live examples. 
If I do something like that, I am getting error 
const Autocomplete: React.FC<any> = () => {
    const [items, setItems] = useState<any>("https://freetestapi.com/api/v1/countries");
    const dataAuto = {
        "apiUrl": "https://freetestapi.com/api/v1/countries",
        "outputItemTemplate": "<div className=\"cursor-pointer py-2 px-4 w-full text-sm text-gray-800 hover:bg-gray-100 rounded-lg focus:outline-none focus:bg-gray-100 dark:bg-neutral-900 dark:hover:bg-neutral-800 dark:text-neutral-200 dark:focus:bg-neutral-800\" data-hs-combo-box-output-item><div className=\"flex justify-between items-center w-full\"><div><div data-hs-combo-box-output-item-field=\"name\" data-hs-combo-box-search-text data-hs-combo-box-value></div></div><div className=\"hidden hs-combo-box-selected:block\"><svg className=\"flex-shrink-0 size-3.5 text-blue-600 dark:text-blue-500\" xmlns=\"http:.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" stroke=\"currentColor\" strokeWidth=\"2\" strokeLinecap=\"round\" stroke-linejoin=\"round\"><polyline points=\"20 6 9 17 4 12\"></polyline></svg></div></div></div>"
      };
    return (
        <>
    <div className="relative" data-hs-combo-box={dataAuto}>
    <div className="relative">
        <input className="py-3 px-4 block w-full border-gray-200 rounded-lg text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-neutral-400 dark:placeholder-neutral-500 dark:focus:ring-neutral-600" type="text" data-hs-combo-box-input="" />
            <div className="absolute top-1/2 end-3 -translate-y-1/2" data-hs-combo-box-toggle="">
                <svg className="flex-shrink-0 size-3.5 text-gray-500 dark:text-neutral-500" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
                    <path d="m7 15 5 5 5-5"></path>
                    <path d="m7 9 5-5 5 5"></path>
                </svg>
            </div>
        </div>
    <div className="absolute z-50 w-full max-h-72 p-1 bg-white border border-gray-200 rounded-lg overflow-hidden overflow-y-auto [&::-webkit-scrollbar]:w-2 [&::-webkit-scrollbar-thumb]:rounded-full [&::-webkit-scrollbar-track]:bg-gray-100 [&::-webkit-scrollbar-thumb]:bg-gray-300 dark:[&::-webkit-scrollbar-track]:bg-neutral-700 dark:[&::-webkit-scrollbar-thumb]:bg-neutral-500 dark:bg-neutral-900 dark:border-neutral-700"
        style={{ display: "none" }} data-hs-combo-box-output=""></div>
</div>
        </>
    );
};
export default Autocomplete;

Error: 
Uncaught SyntaxError: "[object Object]" is not valid JSON
    at JSON.parse (<anonymous>)
    at preline.js:41:19255
    at NodeList.forEach (<anonymous>)
    at e3.autoInit (preline.js:41:19050)
    at preline.js:41:22859

**Comment by Aux9** [comment]
Hi.. Did u get anything so far? I face same problem the templates here almost like top secret codes, no documentation no help nothing is clear and cannot be changed so easily because we don't understand it clearly :S
